### PR TITLE
Pedantic edits for sh

### PR DESCRIPTION
**Pull request by Artoria2e5**
* Remove all invocations of echo due to the poorly-defined handling of
  backslashes (will show up in paths because user input). Now using printf.
* Do not let C handle \e, because that's non-standard.
* Remove handling of percents in SConscript and make it not hand escapes
  to C.
* Add -- to invocations of external commands to guard against potential
  option-looking filenames.
* The same stuff in sh.c.in.

### CODE REVIEW

File: lib/SConscript [hunk 1]
```diff
@@ -50,24 +50,20 @@ def build_config_template(target, source, env):
         handle.write(config_h)
 
 
-def prepare_c_source(text, escape_percentages=True):
+def prepare_c_source(text):
     # Prepare the Python source to be compatible with C-strings
     text = text.replace('"', '\\"')
-    text = text.replace('\\n', '\\\\n')
+    text = text.replace('\\', '\\\\')
```
Comment (sahib) [inline lib/SConscript:56]: What's that reason for this change? It breaks compilation. The issue is also visible on Jenkins: https://travis-ci.org/github/sahib/rmlint/jobs/699617161

Comment (Artoria2e5) [reply lib/SConscript:56]: The idea is to don't leak any escapes to C at all. I effed up when I let the quotes run first.

**Comment by sahib** [comment]
Please change the PR target to ``develop``, then I will merge it.

**Comment by Artoria2e5** [comment]
Done.

**Pull request merged**